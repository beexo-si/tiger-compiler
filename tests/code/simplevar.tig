/*TIGER_ARGS=-canon -code*/
let
    var s:="asdasdasd"
    var x:=1234
    var u:=x+8
in
    u-1
end
