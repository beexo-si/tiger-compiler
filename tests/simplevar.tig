/*TIGER_ARGS=-arbol -ir*/
let
    var x:=0
in
    x
end
