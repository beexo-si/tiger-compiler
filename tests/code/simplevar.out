bien!
------Canon------
;;-END-PROC--:
L0:
	.string "asdasdasd"
L1:	.long 9
;;--END-STRS--:
--FRAME L0:
LABEL L3
MOVE(
 TEMP T0,
 NAME L1)
MOVE(
 TEMP T1,
 CONST 1234)
MOVE(
 TEMP T2,
 BINOP(PLUS,
  TEMP T1,
  CONST 8))
EXP(
 BINOP(MINUS,
  TEMP T2,
  CONST 1))
JUMP( NAME L2)
LABEL L2
;;-END-FRAME-:
------Code------
--FRAME L0:
LABEL: L3
OPER: mov %d0, L1 D:[T3] S:[]
MOVE: mov %d0, %s0 D:T0 S:T3
OPER: mov %d0, 1234 D:[T4] S:[]
MOVE: mov %d0, %s0 D:T1 S:T4
MOVE: mov %d0, %s0 D:T5 S:T1
OPER: add %d0, 8 D:[T5] S:[T5]
MOVE: mov %d0, %s0 D:T2 S:T5
MOVE: mov %d0, %s0 D:T6 S:T2
OPER: sub %d0, 1 D:[T6] S:[T6]
OPER: jmp %j0 D:[] S:[] J:[L2]
LABEL: L2
;;-END-FRAME-:
yes!!
